internal/standalone/standalone.go

```diff
@@ -17,113 +18,153 @@ import (
-	PID        int
-	Standalone bool
-	Dir        string
-	Debugger   bool
+var (
+	standaloneEnabled = flag.Bool("standalone", false, "should this run standalone")
+)
+
+func NewServerSettings(address, dir string) ServerSettings {
+	return ServerSettings{
+		Address: address,
+		Dir:     dir,
+	}
 }
 
-// StandaloneAddressFilePath returns the path to the standalone.txt file, which contains the standalone GRPC address
-func (a Args) StandaloneAddressFilePath() string {
-	return filepath.Join(a.Dir, "standalone.txt")
+type ServerSettings struct {
+	Address string
+	Dir     string
 }
 
-// StandalonePIDFilePath returns the path to the pid.txt file, which contains the standalone GRPC's server PID
-func (a Args) StandalonePIDFilePath() string {
-	return filepath.Join(a.Dir, "pid.txt")
+type ClientSettings struct {
+	TargetAddress string
+	TargetPID     int
 }
 
-func getInfo(id string, executable string, standalone, debug bool) (Args, error) {
-	var info Args
-	info.Standalone = standalone
-
-	// VsCode names the file "__debug_bin"
-	vsCodeDebug := strings.HasPrefix(filepath.Base(executable), "__debug_bin")
-	// GoLand places it in:
-	//  Linux: /tmp/GoLand/___%d%(CONFIGNAME)s_pkg
-	//  Mac OS X: /private/var/folders/lx/XXX/T/GoLand/___%d%(CONFIGNAME)s_pkg
-	//  Windows: C:\Users\USER\AppData\Local\Temp\GoLand\___%d%(CONFIGNAME)s_pkg.exe
-	goLandDebug := strings.Contains(executable, filepath.Join("GoLand", "___"))
-	if standalone && (vsCodeDebug || goLandDebug || debug) {
-		info.Debugger = true
-		js, err := findPluginJSON(executable)
+// ServerModeEnabled returns true if the plugin should run in standalone server mode.
+func ServerModeEnabled(pluginID string) (ServerSettings, bool) {
+	flag.Parse() // Parse the flags so that we can check values for -standalone
+
+	if *standaloneEnabled {
+		s, err := serverSettings(pluginID)
 		if err != nil {
-			return info, err
+			log.Printf("Error: %s", err.Error())
+			return ServerSettings{}, false
 		}
-		executable = js
+		return s, true
 	}
-	info.Dir = filepath.Dir(executable)
+	return ServerSettings{}, false
+}
 
-	// Determine standalone address + PID
-	var err error
-	info.Address, err = getStandaloneAddress(id, info)
+// ClientModeEnabled returns standalone server connection settings if a standalone server is running and can be reached.
+func ClientModeEnabled(pluginID string) (ClientSettings, bool) {
+	s, err := clientSettings(pluginID)
 	if err != nil {
-		return info, err
+		return ClientSettings{}, false
+	}
+
+	return s, true
+}
+
+// RunDummyPluginLocator runs the standalone server locator that Grafana uses to connect to the standalone GRPC server.
+func RunDummyPluginLocator(address string) {
+	fmt.Printf("1|2|tcp|%s|grpc\n", address)
+	t := time.NewTicker(time.Second * 10)
+
+	for ts := range t.C {
+		fmt.Printf("[%s] using address: %s\n", ts.Format("2006-01-02 15:04:05"), address)
 	}
-	info.PID, err = getStandalonePID(info)
+}
+
+func serverSettings(pluginID string) (ServerSettings, error) {
```

Comment:
Nit: I would add docstrings here as this has the side effect of creating the standalone file and that can be confusing. By just looking at the name of function it was not immediate to me that this function creates the standalone file.
Something like:
```
serverSettings finds the plugin root directory, creates a new standalone
address file in it and returns a ServerSettings representing it.
```

Reply:
This doesn't actually create the file - that behaviour is the same as before (the caller does that through the exported func in `standalone`). Maybe it's confusing because the function in here is called `createStandaloneAddress`?

Reply:
Ah yes, sorry! I got confused :sweat_smile: